src/backend/access/external/hd_work_mgr.c

```diff
@@ -768,6 +769,16 @@ make_allocation_output_string(List *segment_fragments)
 	initStringInfo(&segwork);
 	appendStringInfoString(&segwork, SEGWORK_PREFIX);
 	
+	if (enable_secure_filesystem)
```

Comment:
Currently the implementation to add the dfs_address through the external framework is not triggered in the readable case. Ideally, yes, the same implementation would apply to both. So we would like to get this fix in, tested, and then work on consolidating them.